File: VB/Controllers/HomeController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc
Imports Example.Models

Namespace Example.Controllers
	Public Class HomeController
		Inherits Controller
		Public Function Index() As ActionResult
			ViewBag.Message = "Welcome to DevExpress Extensions for ASP.NET MVC!"

			Return View(HomeController.GetObjects())
		End Function

		Public Function GridViewPartial() As ActionResult
			Return PartialView("GridViewPartial", HomeController.GetObjects())
		End Function

		<HttpGet> _
		Public Function CustomAction(ByVal id As Int32) As String
			Return String.Format("A row with the {0} key was selected", id)
		End Function

		Public Shared Function GetObjects() As List(Of MyObject)
			Dim list As New List(Of MyObject) (New MyObject() {New MyObject With {.Key = 1, .Name = "Name 1"}, New MyObject With {.Key = 2, .Name = "Test 2"}, New MyObject With {.Key = 3, .Name = "Something 3"}})

			Return list
		End Function
	End Class
End Namespace

File: CS/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using Example.Models;

namespace Example.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            ViewBag.Message = "Welcome to DevExpress Extensions for ASP.NET MVC!";

            return View(HomeController.GetObjects());
        }

        public ActionResult GridViewPartial() {
            return PartialView("GridViewPartial", HomeController.GetObjects());
        }

        [HttpGet]
        public String CustomAction(Int32 id) {
            return String.Format("A row with the {0} key was selected", id);
        }

        public static List<MyObject> GetObjects() {
            List<MyObject> list = new List<MyObject> {
                new MyObject { Key = 1, Name = "Name 1" },
                new MyObject { Key = 2, Name = "Test 2" },
                new MyObject { Key = 3, Name = "Something 3" }
            };

            return list;
        }
    }
}
